compiler/rustc_ast/src/visitors.rs

```diff
@@ -269,41 +285,44 @@ macro_rules! make_ast_visitor {
-                walk_where_predicate(self, where_predicate)
-            }
-
-            fn visit_trait_ref(&mut self, tr: ref_t!(TraitRef)) -> result!() {
-                walk_trait_ref(self, tr)
-            }
+            make_visit!{Crate, visit_crate, walk_crate}
+            make_visit!{AnonConst, visit_anon_const, walk_anon_const}
+            make_visit!{FormatArgs, visit_format_args, walk_format_args}
+            make_visit!{InlineAsm, visit_inline_asm, walk_inline_asm}
+            make_visit!{InlineAsmSym, visit_inline_asm_sym, walk_inline_asm_sym}
+            make_visit!{Generics, visit_generics, walk_generics}
+            make_visit!{ClosureBinder, visit_closure_binder, walk_closure_binder}
+            make_visit!{WherePredicate, visit_where_predicate, walk_where_predicate}
+            make_visit!{TraitRef, visit_trait_ref, walk_trait_ref}
+            make_visit!{PolyTraitRef, visit_poly_trait_ref, walk_poly_trait_ref}
+            make_visit!{Label, visit_label, walk_label}
+            make_visit!{MacCall, visit_mac_call, walk_mac}
+            make_visit!{PathSegment, visit_path_segment, walk_path_segment}
+            make_visit!{GenericArgs, visit_generic_args, walk_generic_args}
+            make_visit!{GenericArg, visit_generic_arg, walk_generic_arg}
+            make_visit!{AssocItemConstraint, visit_assoc_item_constraint, walk_assoc_item_constraint}
+            make_visit!{Attribute, visit_attribute, walk_attribute}
+            make_visit!{Visibility, visit_vis, walk_vis}
+            make_visit!{FnRetTy, visit_fn_ret_ty, walk_fn_ret_ty}
+            make_visit!{AngleBracketedArgs, visit_angle_bracketed_parameter_data, walk_angle_bracketed_parameter_data}
+            make_visit!{ParenthesizedArgs, visit_parenthesized_parameter_data, walk_parenthesized_parameter_data}
+            make_visit!{ForeignMod, visit_foreign_mod, walk_foreign_mod}
+            make_visit!{MutTy, visit_mt, walk_mt}
+            make_visit!{WhereClause, visit_where_clause, walk_where_clause}
+            make_visit!{EnumDef, visit_enum_def, walk_enum_def}
+            make_visit!{Arm, visit_arm, walk_arm}
+            make_visit!{ExprField, visit_expr_field, walk_expr_field}
+            make_visit!{FieldDef, visit_field_def, walk_field_def}
+            make_visit!{Param, visit_param, walk_param}
+            make_visit!{PatField, visit_pat_field, walk_pat_field}
+            make_visit!{GenericParam, visit_generic_param, walk_generic_param}
+            make_visit!{Variant, visit_variant, walk_variant}
+            make_visit!{P!(Local), visit_local, walk_local}
+            make_visit!{P!(Pat), visit_pat, walk_pat}
+            make_visit!{P!(Expr), visit_expr, walk_expr}
+            make_visit!{P!(Ty), visit_ty, walk_ty}
+            make_visit!{P!(Block), visit_block, walk_block}
+            make_visit!{P!(FnDecl), visit_fn_decl, walk_fn_decl}
```

Comment:
Should we remove the `P!` from all these?

Reply:
I put these `P!` to maintain all function signatures the same, thus avoiding changes on implementers. I don't know why the original MutVisitor received &mut P<T> instead of &mut T for these types, but I just kept what was being done.